docs/source/daml/intro/2_Scenario.rst

```diff
@@ -0,0 +1,146 @@
+
+Archiving Contracts
+-------------------
+
+Archival of contracts works just like creation, but using ``archive`` instead of ``create``. Whilst ``create`` takes an instance of a template, ``archive`` takes a reference to a contract.
+
+Contract references have type ``ContractId a`` in DAML, where ``a`` is a *type parameter* representing the type of contract that the id refers to. A reference to a ``Token`` would be a ``ContractId Token``, for example.
+
+To ``archive`` the Token Alice has created, you need to get a handle on its contract id. In scenarios, this is again done using ``<-`` notation, because the contract ID needs to be retrieved from the ledger. How this works is discussed in :doc:`5_Restrictions`. The scenario first checks that Bob cannot archive Alice's Token and then Alice successfully archives it.
+
+.. literalinclude:: daml/Intro_2_Scenario.daml
+  :language: daml
+  :start-after: -- TOKEN_TEST_3_BEGIN
+  :end-before: -- TOKEN_TEST_3_END
+
+Exploring the Ledger
+--------------------
+
+The resulting scenario view is empty as there are no contracts left on the ledger. However, you may want to see the history of the ledger to see how you got to that state. A simple way to do so is to tick the "Show archived" box at the top of the ledger view:
+
+.. figure:: images/2_Scenario/archived.png
+
+You can see that there was a ``Token`` contract, which is now archived, indicated both in the ``status`` column as well as with a strikethrough. Clicking on the adjacent "Show transaction view" button leads you to a view showing the entire transaction graph.
+
+.. figure:: images/2_Scenario/tx_graph.png
+
+In the IDE scenario runner, committed transactions are numbered sequentially. The lines starting with ``TX`` indicate that there are three commits, with ids ``#0``, ``#1``, and ``#2``. These correspond to the three ``submit`` and ``submitMustFail`` statements in the scenario.
+
+Transaction ``#0`` has one *sub-transaction* ``#0:0``, which the arrow indicates is a ``create`` of a ``Token``. Identifiers ``#X:Y`` can be read as ``commit X, sub-transaction Y``. All transactions have this format in the scenario runner. However, this is a testing feature. Transaction and Contract IDs should be considered opaque in general. The lines above and below ``create Intro_2_Scenario:Token`` give additional information:
+
+- ``consumed by: #2:1`` indicates that the contract is archived in sub-transaction ``1`` of commit ``2``.
+- ``referenced by #2:0, #2:1`` tells you that the contract was used in other transactions and lists their ids.
+- ``known to (since): 'Alice' (#0)`` tells you who knows about the contract. The fact that ``'Alice'`` appears in the list is equivalent to a ``x`` in the tabular view. The ``(#0)`` gives you the additional information that ``Alice`` learned about the contract in commit ``#0``.
+- Everything following ``with`` shows the create arguments.
+
+Exercises
+---------
+
+1. Write a template for a second type of Token.
```

Comment:
Does this mean they need to write a second Token contract? Might be worth being explicit.

Reply:
How is this not explicit?

Reply:
Sorry, somehow I read this as "scenario" rather than "template", not sure how.